source/features/previous-version.tsx

```diff
@@ -0,0 +1,124 @@
+import * as pageDetect from 'github-url-detection';
+
+import React from 'dom-chef';
+
+import cache from 'webext-storage-cache';
+
+import features from '../feature-manager';
+import * as api from '../github-helpers/api';
+import observe from '../helpers/selector-observer';
+
+import GitHubURL from '../github-helpers/github-url';
+
+const getHistoryOids = cache.function('file-history', async (branch: string, filePath: string): Promise<string[] | false> => {
```

Comment:
Not by a computer but take a look at the initial implementation of this feature for how the api code should be done.

https://github.com/refined-github/refined-github/commit/b01263379d091df018286360968f48cadfe43a86

Reply:
Thank you!
I searched https://github.com/fregante/webext-storage-cache and now know how to use it correctly.

I also find the API calling always based on `refined-github/refined-github` and fixed.

The function needs `owner`, `repository`, `branch` and `filePath`.
So I make the cache key is combining of these values.

Reply:
I think Yakov is referring to the lack of "cache name" here. They key is being set to the URL, but it's missing `previous-versions:`

Reply:
I just noticed that it's been fixed in your last commit